lesson_6/alexandra_nechita/6thHomework_Python.py

```diff
@@ -0,0 +1,141 @@
+from collections import defaultdict
+
+# function that prepares the dataset
+
+def prepare_dataset (raw_data):
+    description = ['Country', ('2011 ', '2012 ', '2013 ', '2014 ', '2015 ', '2016 ', '2017 ', '2018 ', '2019 ')]
+    dataset_dict = defaultdict (list)
+    
+#Varianta 1 - with no use of the description list
+    
+#     for country, coverage in raw_data:
+#         year=2011
+#         for index in coverage:
+#             dataset_dict[country].append ({'year':year, 'coverage':index})
+#             year=year+1
+#     return dataset_dict
+
+
+#Varianta2 - using description list, multiple for loops
+
+#     for country, coverage in raw_data:
+#         for index in reversed(coverage):
+#             for year in reversed(description[1]):
+#                 dataset_dict[country].append ({'year':year, 'coverage':index})
+#     return dataset_dict
+
+#Varianta3 - using description list, only 2 for loops
+
+    for country, coverage in raw_data:
+        year_index=len(description[1])-1
```

Comment:
Solutia asta mi se pare cea mai ok, sa iterezi prin 2 liste in acelasi timp. Avem in python o varianta mai usoara decat sa incrementam manual indexul. Uita-te peste `zip()` https://docs.python.org/3/library/functions.html#zip. `description[1]` si `coverage` au aceisi lungime si pot fi iterate simultan cu `zip`